lang/funcs/sensitive.go

```diff
@@ -0,0 +1,65 @@
+package funcs
+
+import (
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+)
+
+// SensitiveFunc returns a value identical to its argument except that
+// Terraform will consider it to be sensitive.
+var SensitiveFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name:             "value",
+			Type:             cty.DynamicPseudoType,
+			AllowUnknown:     true,
+			AllowNull:        true,
+			AllowMarked:      true,
+			AllowDynamicType: true,
+		},
+	},
+	Type: func(args []cty.Value) (cty.Type, error) {
+		// This function only affects the value's marks, so the result
+		// type is always the same as the argument type.
+		return args[0].Type(), nil
+	},
+	Impl: func(args []cty.Value, retType cty.Type) (ret cty.Value, err error) {
+		return args[0].Mark("sensitive"), nil
```

Comment:
By this, are you suggesting that it should be an error to pass an already-sensitive value to `sensitive`?

I feel a bit ambivalent about that, because I can see arguments on both sides:
* Making it an error will give better feedback about unnecessary calls to `sensitive` for values that are already "naturally" sensitive anyway, which would avoid a situation where folks just start writing `sensitive` all over "just to make sure".
* But on the other hand, perhaps authors would _like_ to add annotations "just to make sure", particularly for awkward situations like where the sensitivity of an attribute changed between module or provider versions and so the user wants to force the sensitive annotation for the versions that don't already have it naturally.

With that said, I don't feel a strong opinion either way, so if your opinion is strong then I'm happy to go with it, and if you're also ambivalent then making it an error would be the safer option for now because we can always remove the error later if we find a good reason to do so.

Reply:
No, I meant "don't mark it twice" as in add an `if` statement. I've been able to/seen double marking happen (it has been the source of bugs). It is probably root-related to my naive using `"sensitive"` as the mark rather than using another type, but it is what it is at the moment. I don't think it should be an error.

Reply:
Oh, interesting... okay. I can see that this would happen if some other code happened to use a string other than `"sensitive"` because internally the marks are a map with the marks as keys.... so I understand that the issue you want to guard against here is that the rest of the system uses `IsMarked` as the definition of "sensitive", regardless of specifically which mark it said, and so calling `.Mark("sensitive")` here could indeed, in that case, end up with the value being doubly marked.

I'll change this to unmark the value first, before applying the `"sensitive"` mark, so that this function is always consistent in what it returns regardless of what other mark strings other parts of Terraform might produce.